homeassistant/components/sia/alarm_control_panel.py

```diff
@@ -0,0 +1,285 @@
+            "Loading last state: %s",
+            state.state if state is not None and state.state is not None else "None",
+        )
+        if (
+            state is not None
+            and state.state is not None
+            and state.state
+            in [
+                STATE_ALARM_ARMED_AWAY,
+                STATE_ALARM_ARMED_CUSTOM_BYPASS,
+                STATE_ALARM_ARMED_NIGHT,
+                STATE_ALARM_DISARMED,
+                STATE_ALARM_TRIGGERED,
+                STATE_UNKNOWN,
+            ]
+        ):
+            self.state = state.state
+        else:
+            self.state = None
+        async_dispatcher_connect(self.hass, DATA_UPDATED, self.async_write_ha_state)
+        self.setup_sia_alarm()
+
+    def setup_sia_alarm(self):
+        """Run the setup of the alarm control panel."""
+        self.assume_available()
+        self._unsub = self.hass.bus.async_listen(
+            self._event_listener_str, self.async_handle_event
+        )
+        self.async_on_remove(self._async_sia_on_remove)
+
+    @callback
+    def _async_sia_on_remove(self):
+        """Remove the unavailability and event listener."""
+        if self._remove_unavailability_tracker:
+            self._remove_unavailability_tracker()
+        if self._unsub:
+            self._unsub()
+
+    async def async_handle_event(self, event: Event):
+        """Listen to events for this port and account and update states.
+
+        If the port and account combo receives any message it means it is online and can therefore be set to available.
+        """
+        if not self.enabled:
+            return
+        sia_event = SIAEvent.from_dict(event.data)  # pylint: disable=no-member
+        new_state = CODE_CONSEQUENCES.get(sia_event.code)
+        if int(sia_event.ri) == self._zone:
+            if new_state is not None:
+                self.state = new_state
+            self._attr.update(get_attr_from_sia_event(sia_event))
+        self.assume_available()
+        self.async_write_ha_state()
+
+    @property
+    def name(self) -> str:
+        """Get Name."""
+        return self._name
+
+    @property
+    def ping_interval(self) -> int:
+        """Get ping_interval."""
+        return str(self._ping_interval)
+
+    @property
+    def state(self) -> str:
+        """Get state."""
+        return self._state
+
+    @property
+    def account(self) -> str:
+        """Return device account."""
+        return self._account
+
+    @property
+    def unique_id(self) -> str:
+        """Get unique_id."""
+        return self._unique_id
+
+    @property
+    def available(self) -> bool:
+        """Get availability."""
+        return self._is_available
+
+    @property
+    def device_state_attributes(self) -> dict:
+        """Return device attributes."""
+        return self._attr
+
+    @property
+    def should_poll(self) -> bool:
+        """Return False if entity pushes its state to HA."""
+        return False
+
+    @state.setter
```

Comment:
We should not set the state from outside the entity directly.  If you need to change the state from the outside, use the dispatcher

Reply:
this is not for setting the state externally, but because there is some logic in the state setting that I want to keep there, instead of putting it in the event handler function, either works but this makes more sense to me.